### ISSUE TITLE
var and val keyword

### ISSUE THREAD

**Issue by Vayvess** [MaVdbussche/nozc#15]
Since "var" and "val" looks the same, they can easily be mistaken into their opposite.
Why not replace "var" keyword by "mut" as Rust does ?

**Comment by MaVdbussche** [comment]
This is something that hadn't crossed my mind. Those keywords were brought from JS and Scala without much questioning.

I am not very familiar with Rust, but another option could be to use `let` and `let mut`, which would effectively make all variables immutable by default, except if stated otherwise. This would in fact be more faithful to the way Oz does things, so to speak.

Thanks for this input !

**Comment by pnrao** [comment]
Instead of using the keyword `val` for CTM's variables, and `var` for CTM's, how about staying close to CTM's linguistic conventions? So, things that are called variables in CTM could be `var` in nozc, and things that are called cells in CTM could be called `cel` or `cell` in nozc.

Unless Profs. Van Roy & Haridi plan on releasing a new edition of CTM for nozc, students will be reading the original for the time being. Switching between the two different frames of mind will be easier for new students if the conventions have a good degree of correspondence between ozc and nozc.

**Comment by MaVdbussche** [comment]
Your point about the book is very relevant. However, my position is the following : we are designing a language based on Oz, and not around the documentation/resources/courses that have made use of Oz over the years. By that I mean that the material should be adapted to the new syntax when it will be officially released. I think that, by the time the language is mature enough to be used in courses, the material will have been adapted to match it, so to speak. I need however to discuss this with Prof. Van Roy that have his opinion.
Regardless of this, I actually like the keyword `cell` a lot. Despite the opinion I gave about the *essence* of your point -which, I want to stress this, might change and is really not the only one taken into account in this whole project- I actually like the keyword `cell` a lot; it can't hurt to try and stay closer to the original syntax, which is the primary goal after all.

TL;DR : even though I don't think we should constrain ourselves too much by matching the existing material, in this case it makes a lot of sense and I find it quite elegant

Thank you for this input !
